pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+    def _fake_getclass(self, namespace, **params):
+        """
+        Implements a mock equivalent of
+        :meth:`~pywbem.WBEMConnection.GetClass
+
+        Retrieve a CIM class from the local repository.
+
+        For a description of the parameters, see
+        :meth:`pywbem.WBEMConnection.GetClass`.
+        """
+        self._validate_classes_namespace(namespace)
+        cname = params['ClassName'].classname
+
+        cc = self._get_class(cname, namespace, **params)
+        # TODO need to get away from having to put list around single return
+        return self._make_tuple([cc])
+
+    def _fake_createclass(self, namespace, **params):
+        """
+        Implements a mock server reactor for
+
+        :meth:`~pywbem.WBEMConnection.CreateClass`
+
+        Creates a new class in the repository.  Nothing is returned.
+        Emulates WBEMConnection.CreateClass(...))
+
+        if the class repository for this namespace does not
+        exist, this method creates it.
+
+        Returns:
+            None
+        Exceptions:
+            CIMError
+        """
+        new_class = params['NewClass']
+        if namespace not in self.classes:
+            self.classes[namespace] = NocaseDict({})
+
+        if not isinstance(new_class, CIMClass):
+            CIMError(CIM_ERR_INVALID_PARAMETER, 'NewClass not valid class\n%s'
+                     % new_class)
+
+        if new_class.superclass:
+            try:
+                _ = self._get_class(new_class.superclass,  # noqa: F841
+                                    namespace=namespace,
+                                    LocalOnly=True,
+                                    IncludeQualifiers=False)
+            except CIMError as ce:
+                if ce.args[0] == CIM_ERR_NOT_FOUND:
+                    ce.args = (CIM_ERR_INVALID_SUPERCLASS, new_class.superclass)
+                    raise
+                else:
+                    raise
+
+        if new_class.classname in self.classes[namespace]:
+            raise CIMError(CIM_ERR_ALREADY_EXISTS, new_class.classname)
+
+        self.classes[namespace][new_class.classname] = new_class
+
+    def _fake_deleteclass(self, namespace, **params):
+        """
+        Implements a mock server reactor for
+        :meth:`~pywbem.WBEMConnection.DeleteClass`
+
+        Delete a class in the class repository if it exists.
+        Emulates WBEMConnection.DeleteClass(...))
+
+        This is simplistic in that it ignores issues like existing
+        subclasses and existence of instances.
+
+        Nothing is returned.
+
+        Exceptions:
+            CIMError CIM_ERR_NOT_FOUND
+        """
+        self._validate_classes_namespace(namespace)
+
+        cname = params['ClassName'].classname
+
+        try:
+            del self.classes[namespace][cname]
+        except KeyError as ke:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Cannot Delete %s. Not in '
+                           'repository %s' % (cname, ke))
```

Comment:
1. Showing `ke` in the exception message is not necessary, we know that the problem is that the class was not found.
2. make the exception message consistent with others, by saying that the class was not found.
3. Mention the name of the namespace in which the class was looked up, in the exception message.

Reply:
Done